File: sms-api-java/authservice/src/main/java/com/example/mydemoproject/authservice/service/UserService.java
package com.example.mydemoproject.authservice.service;

import com.example.mydemoproject.authservice.model.User;

import java.util.List;

public interface UserService {

    default User saveUser(User user) {
        long id = (long) Math.random();
        return saveUser(user);
    }
    List<User> getAllUsers();
    User getUserById(long id);
    void deleteUser(long id);
    User updateUser(User user);
    User findByUsername(String username);
}

File: sms-api-kotlin/authservice/src/main/kotlin/com/example/authservice/service/UserService.kt
package com.example.authservice.service

import com.example.userservice.model.User


interface UserService {
    fun saveUser(user: User): User {
        val id = Math.random().toLong()
        return saveUser(user)
    }
    fun getAllUsers(): List<User>
    fun getUserById(id: Long): User
    fun deleteUser(id: Long)
    fun updateUser(id: Long, user: User): User
    fun findByUsername(username: String): User
}


